### ISSUE TITLE
improve docker pipe tests

### ISSUE THREAD

**Issue by caarlos0** [goreleaser/goreleaser#379]
`go test` may fail in the `docker` pipe if the user is not logged in with a docker account with permission into the `goreleaser` org, as reported in [here](https://gophers.slack.com/archives/C3T1JU24B/p1506799845000010).

Let's fix this!

**Comment by jorinvo** [comment]
Error looks like this:
```
command [go test -cover -covermode atomic -coverprofile /var/folders/km/kqlfxy2j64dd_9gqb373c0hm0000gn/T/gotestcover-526170983 github.com/goreleaser/goreleaser/pipeline/docker -run . -timeout=2m]: exit status 1
2017/10/01 19:57:53  info building docker image     image=goreleaser/test_run_pipe:1.0.0
2017/10/01 19:57:56  info tagging docker image      image=goreleaser/test_run_pipe:1.0.0 tag=goreleaser/test_run_pipe:latest
2017/10/01 19:57:57  info pushing docker image      image=goreleaser/test_run_pipe:1.0.0
--- FAIL: TestRunPipe (8.49s)
        Error Trace:    docker_test.go:62
	Error:  	Received unexpected error exit status 1
			failed to push docker image: 
			The push refers to a repository [docker.io/goreleaser/test_run_pipe]
			5602e019ad66: Preparing
			denied: requested access to the resource is denied
			
			github.com/goreleaser/goreleaser/pipeline/docker.dockerPush
				github.com/goreleaser/goreleaser/pipeline/docker/_test/_obj_test/docker.go:186
			github.com/goreleaser/goreleaser/pipeline/docker.publish
				github.com/goreleaser/goreleaser/pipeline/docker/_test/_obj_test/docker.go:129
			github.com/goreleaser/goreleaser/pipeline/docker.process
				github.com/goreleaser/goreleaser/pipeline/docker/_test/_obj_test/docker.go:113
			github.com/goreleaser/goreleaser/pipeline/docker.doRun
				github.com/goreleaser/goreleaser/pipeline/docker/_test/_obj_test/docker.go:66
			github.com/goreleaser/goreleaser/pipeline/docker.Pipe.Run
				github.com/goreleaser/goreleaser/pipeline/docker/_test/_obj_test/docker.go:42
			github.com/goreleaser/goreleaser/pipeline/docker.TestRunPipe
				/Users/jorin/go/src/github.com/goreleaser/goreleaser/pipeline/docker/docker_test.go:62
			testing.tRunner
				/usr/local/Cellar/go/1.9/libexec/src/testing/testing.go:746
			runtime.goexit
				/usr/local/Cellar/go/1.9/libexec/src/runtime/asm_amd64.s:2337
		
FAIL
```

**Comment by mattstratton** [comment]
I'm not sure that #381 fixes this - it's not about having docker, it's about having credentials that can push to that dockerhub org...

**Comment by jorinvo** [comment]
No #381 was not a fix for this. Just marked it as related. This issue here is still open and unsolved :/

**Comment by caarlos0** [comment]
just opened a PR that may fix this: #395

**Comment by jorinvo** [comment]
Thank you @caarlos0!
